### PR TITLE
feat: [CDS-46922]: fixes for supporting org/acc level ser/env

### PR DESCRIPTION
**Pull request by hingerabhinav**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42198)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java [hunk 1]
```diff
@@ -576,6 +577,7 @@ public ResponseDTO<EnvironmentInputsMergedResponseDto> mergeEnvironmentInputs(
         @io.swagger.v3.oas.annotations.responses.
         ApiResponse(description = "Upsert ( Create/Update )  a Service Override in an Environment.")
       })
+  @ExposeInternalException(withStackTrace = true)
```
Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java:580]: what is the purpose of this ? Looks ike this was meant for interservice calls

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java [hunk 2]
```diff
@@ -511,7 +511,17 @@ void validatePresenceOfRequiredFields(Object... fields) {
     Lists.newArrayList(fields).forEach(field -> Objects.requireNonNull(field, "One of the required fields is null."));
   }
 
-  private void setName(Environment requestEnvironment) {
+  private void modifyEnvironmentRequest(Environment requestEnvironment) {
+    // create scoped environment
+    String[] envRefSplit = StringUtils.split(requestEnvironment.getIdentifier(), ".", MAX_RESULT_THRESHOLD_FOR_SPLIT);
```
Comment (yogesh-chauhan) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java:516]: This splitting logic is already done by IdentifierRefHelper.getIdentifierRef

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java [hunk 3]
```diff
@@ -511,7 +511,17 @@ void validatePresenceOfRequiredFields(Object... fields) {
...
+    if (envRefSplit != null && envRefSplit.length == 2) {
```
Comment (yogesh-chauhan) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java:517]: Also, do we really need to support refs while CRUD ? Refs should be supported mostly from other entities which refer to the environment

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

File: 125-cd-nextgen/src/main/java/io/harness/cdng/service/steps/ServiceStepsHelper.java [hunk 4]
```diff
@@ -58,7 +60,7 @@ public class ServiceStepsHelper {
   @Inject @Named("PRIVILEGED") private AccessControlClient accessControlClient;
   @Inject EntityDetailProtoToRestMapper entityDetailProtoToRestMapper;
 
-  void checkForVariablesAccessOrThrow(Ambiance ambiance, NGServiceConfig serviceConfig) {
+  void checkForVariablesAccessOrThrow(Ambiance ambiance, NGServiceConfig serviceConfig, String serviceRef) {
```
Comment (hingerabhinav) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/service/steps/ServiceStepsHelper.java:63]: check existing methods usage

**Pull request closed without merging**